### PR TITLE
build: upgrade to node 12 LTS

### LINKED ISSUE #44021: build: upgrade to node 12 LTS
Follow similar steps to bumping golang version documented in the build README

* [x] Adjust version in Docker image ([source](./builder/Dockerfile#L218)).
* [x] Rebuild the Docker image and bump the version in builder.sh accordingly ([source](./builder.sh#L6)).
* [x] Bump the default installed version of nodejs in bootstrap-debian.sh ([source](./bootstrap/bootstrap-debian.sh#L9)).
* [ ] Update the `builder.dockerImage` parameter in the TeamCity `Cockroach` project.

### PR DESCRIPTION
**Pull request by dhartunian**
In order to use the latest webpack version we need
to upgrade the node version as well.

This upgrades us to the current LTS version.

Added some documentation on this in README which 
mirrors many of the steps for a go upgrade.

Added a `.node-version` file to the root of the 
repo which helps with using the `nodenv` tool to 
manage locally installed nodes versions.

* [X] Adjust version in Docker image ([source](./builder/Dockerfile#L218)).
* [X] Rebuild the Docker image and bump the version in builder.sh accordingly ([source](./builder.sh#L6)).
* [X] Bump the default installed version of nodejs in bootstrap-debian.sh ([source](./bootstrap/bootstrap-debian.sh#L9)).
* [x] Update the `builder.dockerImage` parameter in the TeamCity `Cockroach` project.

Closes: #44021

Release note (build change): CockroachDB UI is now built with node12

### CODE REVIEW

**Review by vilterp — APPROVED** [review]
Thanks for the readme checklist, `.node-version` and `engines`!

File: build/README.md [hunk 1]
```diff
@@ -82,6 +82,20 @@ You can test the new builder image in TeamCity by using the custom parameters
...
 
+## Updating the nodejs version
+
+Please copy this checklist ino the relevant commit message, with a link
+back to this document:
+
+* [ ] Adjust version in Docker image ([source](./builder/Dockerfile#L218)).
+* [ ] Rebuild the Docker image and bump the version in builder.sh accordingly ([source](./builder.sh#L6)).
+* [ ] Bump the default installed version of nodejs in bootstrap-debian.sh ([source](./bootstrap/bootstrap-debian.sh#L9)).
+* [ ] Update the `builder.dockerImage` parameter in the TeamCity `Cockroach` project.
```
Comment (bobvawter) [inline build/README.md]: Please don't make a copy of the checklist, they might get out of sync.  Please just add a link back to the relevant section.

**Review by bobvawter — APPROVED** [review]
LGTM w/ nit

**Pull request merged**